scipy/stats/_qmc.py

```diff
@@ -1,13 +1,26 @@
 """Quasi-Monte Carlo engines and helpers."""
+from __future__ import annotations
+
 import os
 import copy
 import numbers
 from abc import ABC, abstractmethod
 import math
+from typing import (
+    ClassVar,
+    List,
+    Optional,
+    overload,
+    TYPE_CHECKING,
+)
 import warnings
 
 import numpy as np
 
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from typing import Literal
```

Comment:
```suggestion
    from typing_extensions import Literal
```

Note that the `Literal` import will currently fail for python 3.7 (as it does not exist in python 3.7!).

You'll need either a version check here _or_ an unconditional import from `typing_extensions`, 
the latter which exposes either a backport or a re-export of `Literal`, depending on the python version.

Reply:
Fyi the import will have to remain within the `if TYPE_CHECKING` block as `typing_extensions` is not a runtime dependency.

Reply:
Oh sorry, I misunderstood : 

>  *or* an unconditional import from `typing_extensions`

I though unconditional meant outside the `if TYPE_CHECKING`.
Thank you very much @BvB93

Reply:
It seems that Scipy doesn't have `typing_extensions`. 
The error can be seen [here](https://github.com/scipy/scipy/runs/2595468155?check_suite_focus=true#step:8:651) : 

```
scipy/stats/__init__.py:444: in <module>
    from . import qmc
scipy/stats/qmc.py:234: in <module>
    from ._qmc import *
scipy/stats/_qmc.py:16: in <module>
    from typing_extensions import Literal
E   ModuleNotFoundError: No module named 'typing_extensions'
```
I found a similar issue [here](https://github.com/facebookresearch/cc_net/issues/3). Should I add `typing_extensions` in scipy requirements or something? @rgommers ?

Reply:
The `ModuleNotFoundError` is from the CI run prior to https://github.com/scipy/scipy/pull/13833/commits/136845f59dd725e53d482e106b9939fcebbecb4b. 
The error should not re-appear again, as the module is now no longer imported during runtime.